### ISSUE TITLE
Filters against the same facet field should be disjunctive

### ISSUE THREAD

**Issue by relwell** [clinwiki-org/clinwiki#60]
Selecting state A and state B should search for (state:A OR state:B)

**Comment by williamhoos** [comment]
this is important because people need to be able to select multiple cities, star values, recruiting status

**Comment by relwell** [comment]
This should take 2 hours or less

**Comment by relwell** [comment]
This is completed and deployed to QA.  You'll note that filters within the same category are disjunctive but across categories filters are *conjunctive*.

The counts for each agg will remain the count for *just the search*.

This means that you may be able to select criteria that result in no results by selecting a combination of facets across different categories for which no document matches.

In other words, for fields X, Y, and Z, we can ask for documents where X is in [value a, value b, value c] AND Y is in [value 1, value 2, value 3]. If we have documents where Y has a value of "value 1", but it doesn't match any value for field X, that document will not be visible.

**Comment by williamhoos** [comment]
Before, the agg filtered after each facet was applied.  We should look to see whether we can re-institute that behavior so that filter counts to reflect what is still available.  If too difficult with disjunctive, we should evaluate some way to indicate the issue or remove counts when they are no longer accurate.  Seems almost worse to have wrong counts than none at all.

Let's tackle when addressing #87 for sharing a faceted filter result.

**Comment by relwell** [comment]
I was able to fix this by implementing our own filters for each agg that uses data from all the other aggs we've selected. Searchkick provides a "smart_aggs" option that will do this for you, but that one *does* include the filter for that agg, which screws up the counts.
